tests/PKPSwordDepositTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ * @file tests/PKPSwordDepositTest.php
+ *
+ * Copyright (c) 2014-2020 Simon Fraser University
+ * Copyright (c) 2000-2020 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class PKPSwordDepositTest
+ * @see PKPSwordDeposit
+ *
+ * @brief Tests for the PKPSwordDeposit class.
+ */
+
+require_once __DIR__ . '/../classes/PKPSwordDeposit.inc.php';
+
+class MockSessionManager {
+	function getUserSession() {
+		return new class {
+			function getSessionVar($__) {
+				return 'en_US';
+			}
+		};
+	}
+}
+
+class MockRequest {
+	function getUserVar($param) {
+		return null;
+	}
+
+	function getJournal() {
+		return null;
+	}
+
+	function getSite() {
+		return new class {
+			function getSupportedLocaleNames() {
+				return ['en_US'];
+			}
+
+			function getPrimaryLocale() {
+				return 'en_US';
+			}
+		};
+	}
+
+	function getCookieVar($param) {
+		return 'en_US';
+	}
+}
+
+class TestablePKPSwordDeposit extends PKPSwordDeposit {
+
+	function getPackage (){
+		return $this->_package;
+	}
+
+	function getOutPath () {
+		return $this->_outPath;
+	}
+}
+
+class PKPSwordDepositTest extends PHPUnit\Framework\TestCase {
+
+	static $metsDocPaths = [];
+
+	public static function setUpBeforeClass(): void {
+		define('INDEX_FILE_LOCATION', 'foo');
+		define('BASE_SYS_DIR', dirname(INDEX_FILE_LOCATION));
+		require(BASE_SYS_DIR . '/lib/pkp/includes/functions.inc.php');
+		import('classes.core.Application');
+		$application = new Application();
+		$mockSessionManager = new MockSessionManager();
+		Registry::set('sessionManager', $mockSessionManager);
+		$submissionDAO = $application->getSubmissionDAO();
+		AppLocale::initialize(new MockRequest);
+		foreach ([9, 17] as $submissionId) {
```

Comment:
Are these IDs from the test dataset, or is this stuff mocked?

Reply:
Ah, I see there's a SQL dump included with the plugin. This is based on our test dataset, right? It'll be tough to maintain this way; instead, I'd suggest having Cypress generate the test data before the tests get run. See https://github.com/pkp/oaiJats/blob/master/.travis.yml for an example of this.